File: leetcode/python/38.CountAndSay.py
class Main:

  def countAndSay(self,n):
    result = ""
    for i in range(n):
      if i==0:
        result = "1"
        continue
      result = self.process(result)
    return result
  
  def process(self,s):
    result = ""
    curr = s[0]
    count = 1
    for i in range(1,len(s)):
      if s[i] != curr:
        result += str(count) + str(curr)
        curr = s[i]
        count = 1
        continue
      count +=1
    result += str(count) + "" + str(curr)
    return result

m = Main()
print(m.countAndSay(4))

File: leetcode/c++/38.CountAndSay.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

string process(string s){
  string result = "";
  char curr = s[0];
  int count = 1;
  for (int i = 1; i < s.length(); i ++) {
    if (s[i] != curr) {
      result += (to_string(count) + "") + curr;
      curr = s[i];
      count = 1;
      continue;
    }
    count ++;
  }
  result += (to_string(count) + "") + curr;
  return result;
}

string countAndSay(int n){
  string result="";
  for(int i=0;i<n;i++){
    if(i==0){
      result = "1";
      continue;
    }
    result = process(result);
  }
  return result;
}

int main(){
  cout << countAndSay(4) << endl; 
  return 0;
}